modules/shared_vpc_access/main.tf

```diff
@@ -61,7 +61,7 @@ resource "google_compute_subnetwork_iam_member" "service_shared_vpc_subnet_users
    if "dataflow.googleapis.com" compute.networkUser role granted to dataflow  service account for Dataflow on shared VPC Project if no subnets defined
  *****************************************/
 resource "google_project_iam_member" "service_shared_vpc_user" {
-  for_each = length(var.shared_vpc_subnets) == 0 ? local.active_apis : []
+  for_each = (length(var.shared_vpc_subnets) == 0) && (var.host_project_id != "") ? local.active_apis : []
```

Comment:
I believe what is happening here is in our svpc example https://github.com/terraform-google-modules/terraform-google-project-factory/blob/v8.1.0/examples/shared_vpc/main.tf  we test dynamic host + service project creation. However when you added `var.host_project_id != ""`, `var.host_project_id` is determined dynamically and hence TF complains ` "for_each" value depends on resource attributes that cannot be determined until apply`.

I believe one solution could be modifying the `shared_vpc_access` to take in a boolean `shared_vpc_enabled` variable. Then in the `shared_vpc` module change the `shared_vpc_access` module instantiation to 
```tf
module "shared_vpc_access" {
  ...
  shared_vpc_enabled = true
}
```

In the root module change the `shared_vpc_access` module instantiation to 
```tf
module "shared_vpc_access" {
  ...
  shared_vpc_enabled = var.host_project_id != "" ? true : false
}
```
 This will enable users who want to dynamically provision both host and service projects to use a combination of the root module (for host project) and  `shared_vpc` module (for service projects) as shown [here](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/master/examples/shared_vpc/main.tf) 

/cc @morgante would this preserve historical behavior for both the root and `shared_vpc` modules?

Reply:
Hi @bharathkkb I think that approach makes sense.